internal/provider/networking/network_ip_interface_resource.go

```diff
@@ -107,16 +122,40 @@ func (r *IPInterfaceResource) Schema(ctx context.Context, req resource.SchemaReq
 				},
 				Required: true,
 			},
-			// TODO: Make location fields optionals once other options are supported
 			"location": schema.SingleNestedAttribute{
 				Attributes: map[string]schema.Attribute{
 					"home_node": schema.StringAttribute{
 						MarkdownDescription: "IPInterface home node",
-						Required:            true,
+						Optional:            true,
+						Computed:            true,
 					},
 					"home_port": schema.StringAttribute{
 						MarkdownDescription: "IPInterface home port",
-						Required:            true,
+						Optional:            true,
+						Computed:            true,
+					},
+					"broadcast_domain": schema.SingleNestedAttribute{
+						Attributes: map[string]schema.Attribute{
+							"name": schema.StringAttribute{
+								MarkdownDescription: "Name of the broadcast domain, scoped to its IPspace",
+								Optional:            true,
+								Computed:            true,
+								PlanModifiers: []planmodifier.String{
+									stringplanmodifier.RequiresReplace(),
+								},
+							},
+							"id": schema.StringAttribute{
```

Comment:
We typically don't have UUID in resources other than the top level one. As we only expect users to know the names and not the internal ID.

We do the top level one as computed as we do need a unique value for terraform. 

So this ID can probably be removed for the Resouce. (For the data source it can stay)

Reply:
The ID is optional, so users don't need to use it. However, referencing a broadcast domain by ID has the following advantage: the ID is computed by the `broadcast_domain` resource, it is "unknown" during plan-phase and only has a known value after the broadcast domain exists. Hence, Terraform implicitly understands that the `broadcast_domain` resource needs to be created first, and the interface can only be created *after* the `broadcast_domain`. When referencing a `broadcast_domain` by name, users would need to explicitly add this dependency with the `depends_on` meta-argument.

Hence, I'd suggest to keep the UUID as an optional attribute - do you agree?